pulp_2to3_migration/app/viewsets.py

```diff
@@ -57,6 +61,12 @@ def run(self, request, pk):
         validate = serializer.validated_data.get('validate', False)
         dry_run = serializer.validated_data.get('dry_run', False)
 
+        # find running/waiting migration plugin tasks
+        qs = TaskReservedResourceRecord.objects.filter(resource__resource='pulp_2to3_migration',
+                                                       task__state__in=['running', 'waiting'])
```

Comment:
I believe it's not enough. This was valid in the past and the tasking system took care of that.

The only task with pulp_2to3_migration resource is the first task which triggers migrations. It doesn't wait for other child tasks to be complete.
You need to look at all tasks which are running or waiting and see which task_group they belong to if any. Then check those task groups, if they have any task (even completed ones) with the reserved_resource = pulp_2to3_migration and if so, raise an error.

@dkliban, @dralley , keep me honest here, please.

Reply:
@goosemania Probably correct, yes.

Reply:
I don't think it's necessary to check on reserved_resources. You should be able to just check to see if the task group has any tasks in running or waiting and if the all_tasks_dispatched is True.

Reply:
@dkliban At this point we don't know which task group was "the task group" from the last migration, so we would have to figure that out as well.  I guess you could walk back through the tasks, check the code on each looking for migration tasks, and then grab the task group, and *then* do that.